modules/javafx.base/src/test/java/test/util/memory/JMemoryBuddy.java

```diff
@@ -0,0 +1,299 @@
+                }
+            }
+            for(AssertNotCollectable wRef: toBeNotCollected) {
+                if(wRef.getWeakReference().get() == null) {
+                    toBeNotCollectedMarked.add(wRef);
+                }
+            }
+            createHeapDump();
+            if(toBeNotCollectedMarked.isEmpty()) {
+                throw new AssertionError("The following references should be collected: " + toBeCollectedMarked);
+            } else {
+                throw new AssertionError("The following references should be collected: " + toBeCollectedMarked + " and " + toBeNotCollected.size() + " should not be collected: " + toBeNotCollectedMarked);
+            }
+        }
+
+
+    }
+
+
+    static void createHeapDump() {
+        if(createHeapdump) {
+            try {
+                String dateString = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss").format(new Date());
+                String fileName = "heapdump_jmemb_" + dateString + ".hprof";
+                File outputFolder = new File(outputFolderString);
+                String heapdumpFile = new java.io.File(outputFolder, fileName).getAbsolutePath();
+                System.out.println("Creating Heapdump at: " + heapdumpFile);
+                getHotspotMBean().dumpHeap(heapdumpFile, true);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            System.out.println("No Heapdump was created. You might want to change the configuration to get a HeapDump.");
+        }
+    }
+
+    private static void setMxBeanProxyName(String mxBeanName) {
+        MX_BEAN_PROXY_TYPE = mxBeanName;
+    }
+
+    private static HotSpotDiagnosticMXBean getHotspotMBean() throws IOException {
+        MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+        HotSpotDiagnosticMXBean bean =
+                ManagementFactory.newPlatformMXBeanProxy(server,
+                        MX_BEAN_PROXY_TYPE, HotSpotDiagnosticMXBean.class);
+        return bean;
+    }
+
+    public static interface MemoryTestAPI {
+        /**
+         * After executing the lambda, the provided ref must be collectable. Otherwise an Exception is thrown.
+         * @param ref The reference which should be collectable.
+         */
+        public void assertCollectable(Object ref);
+        /**
+         * After executing the lambda, the provided ref must be not collectable. Otherwise an Exception is thrown.
+         * @param ref The reference which should not be collectable.
+         */
+        public void assertNotCollectable(Object ref);
+
+        /**
+         * The provided reference will be reference hard, so it won't be collected, until memoryTest finishes.
+         * @param ref The reference which should get a hard reference for this test.
+         */
+        public void setAsReferenced(Object ref);
+    }
+
+    static class AssertCollectable {
+        WeakReference<Object> assertCollectable;
```

Comment:
There is method, class member variable and class itself with the same name `assertCollectable`.
I think they should be named differently, like for instance,
1. Class `AssertCollectable` can be named as `CollectableReferernce` or `ReferenceToBeGC`
2. The member variable `assertCollectable` can be named as just `weakRef` or `reference`.
3. The method `assertCollectable` can be named as `checkIfGCed` or `checkIfCollected`

Similar change for `AssertNotCollectable`

Reply:
1.) Main purpose of the class is being easy searchable in tools like VisualVM. CollectableReference isn't correct, because we don't know whether it's collectable. That's what we are checking.
2.) true, but the old name has the benefit to be easier recognizable in tools like VisualVM.
3.) See other comments on the same topic.